Command.php

```diff
@@ -288,7 +288,9 @@ public function update($index, $type, $id, $data, $options = [])
             unset($options["detect_noop"]);
         }
 
-        if ($this->db->dslVersion >= 7) {
+        if ($this->db->dslVersion >= 8) {
+            return $this->db->post([$index, '_update', $id ], $options, Json::encode($body));
```

Comment:
Actually no need to distinguish for version 7 and 8 - new (fixed) syntax works for both well. 
In ES 7 this worked for BC compatability reasons also with ES 6 syntax.
https://www.elastic.co/guide/en/elasticsearch/reference/7.17/removal-of-types.html#_schedule_for_removal_of_mapping_types